tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -6793,12 +6797,30 @@ Status ConvertGraphDefToEngine(
   TF_RETURN_IF_ERROR(statusor.status());
   auto converter = std::move(statusor.ValueOrDie());
 
+  tensorflow::grappler::GrapplerItem grappler_item;
+
+  GraphDef output_graph;
+
+  grappler_item.graph = gdef;
+  if (cluster != nullptr) {
+    tensorflow::grappler::GrapplerItem grappler_item;
+    grappler_item.graph = gdef;
+    tensorflow::grappler::GenericLayoutOptimizer layout_optimizer("NCHW");
+    layout_optimizer.Optimize(cluster, grappler_item, &output_graph);
+
+    grappler_item.graph = output_graph;
+
+    tensorflow::grappler::ConstantFolding const_optimizer(nullptr);
```

Comment:
Note that constant folding can affect how we handle quantization ops. See note here https://github.com/tensorflow/tensorflow/blob/afec6f4d62c1ba96b3899d48d35793dfeb85ab6e/tensorflow/compiler/tf2tensorrt/trt_convert_api.cc#L93-L100

You can use the last arg of ConstantFolding to control folding of quantization ops: https://github.com/tensorflow/tensorflow/blob/afec6f4d62c1ba96b3899d48d35793dfeb85ab6e/tensorflow/core/grappler/optimizers/constant_folding.h#L50